### PR TITLE
Feature/post new hydration

### PR DESCRIPTION
**Pull request by ADavidson02**
## Is this a fix or a feature?
Feature

## What is the change?
the user is now able to post new hydration data to the server 

## Where should the reviewer start?
src/scripts.js line 30

## How should this be tested?
on the main page the dev tools should be open.
Enter an integer in the new hydration input field and press "Enter today's hydration info"
The user's ounces today drank should update to the integer that was used in the input field and
the average ounces should also update. 

## Relevant Ticket(s)
ticket #14 

@Shawntru 
@adavidson02
@sarah01123

### CODE REVIEW

File: src/Sleep.js [hunk 1]
```diff
@@ -17,7 +17,13 @@ class Sleep {
 
   calculateDailySleep(id, date) {
     const findSleepByDate = this.sleepData.find((data) => id === data.userID && date === data.date);
-    return findSleepByDate.hoursSlept;
+    // this.sleepData.forEach(data => {
+    //   typeof data.hoursSlept === "number" ? data.hoursSlept : 0
+    // })
+    // typeof findSleepByDate.hoursSlept === "number" ? findSleepByDate.hoursSlept : 0
+    return this.sleepData.forEach(data => {
```
Comment (Shawntru) [inline src/Sleep.js:24]: What are we trying to return from this function? Returning forEach will always be 'undefined'.

File: src/index.html [hunk 2]
```diff
@@ -46,6 +46,9 @@ <h2 class="sidebar-header-name" id="sidebarName">Name Name</h2>
           <div class="card-today-hydration">
             <p id="hydrationToday"></p>
           </div>
+          <p>Max amount is 128oz</p>
+          <input type="text" class="new-hydration-input" placeholder="Number of ounces" onkeyup="value=value.replace(/[^\d]/g,'')" onkeyup=>
```
Comment (Shawntru) [inline src/index.html:50]: Multiple onkeyup, remove second

File: src/Sleep.js [hunk 3]
```diff
@@ -17,7 +17,13 @@ class Sleep {
 
   calculateDailySleep(id, date) {
     const findSleepByDate = this.sleepData.find((data) => id === data.userID && date === data.date);
-    return findSleepByDate.hoursSlept;
+    // this.sleepData.forEach(data => {
+    //   typeof data.hoursSlept === "number" ? data.hoursSlept : 0
```
Comment (saraho1123) [inline src/Sleep.js:21]: I think we were trying to deal with bad data. I started a new branch that added conditionals to deal with bad data in sleep and activity class.

File: src/fetch.js [hunk 4]
```diff
@@ -25,7 +25,61 @@ let requests = {
       .then(response => response.json())
       .then(data => data.sleepData)
       .catch(error => console.log(error))
-  }
+  },
+  
+  postActivityData(inputId, inputDate, inputSteps, inputMinutesActive, inputStairs) {
```
Comment (saraho1123) [inline src/fetch.js:30]: all the posts look good!

File: src/scripts.js [hunk 5]
```diff
@@ -97,6 +118,69 @@ function getUserById(id, listRepo) {
   return listRepo.getDataFromID(id);
 }
 
+
+function windowOnClick(event) {
+  if (event.target.classList.contains("new-hydration-button")) {
+
```
Comment (Shawntru) [inline src/scripts.js:124]: Whitespace fix

File: src/scripts.js [hunk 6]
```diff
@@ -97,6 +118,69 @@ function getUserById(id, listRepo) {
...
+function getTodaysDate() {
+  var today = new Date();
+  var dd = String(today.getDate()).padStart(2, '0');
+  var mm = String(today.getMonth() + 1).padStart(2, '0'); //January is 0!
+  var yyyy = today.getFullYear();
+  return today = yyyy + '/' + mm + '/' + dd;
+}
+
+
```
Comment (Shawntru) [inline src/scripts.js:143]: Whitespace fix

File: src/scripts.js [hunk 7]
```diff
@@ -97,6 +118,69 @@ function getUserById(id, listRepo) {
...
+  }
+  let postedHydration = requests.postHydrationData(userNowId, todaysDate, newHydrationValue)
+  Promise.all([postedHydration])
+    .then(value => {
+      updatePageHydration (userNowId, todaysDate) 
+  });
+  updateHydrationButton.disabled = true;
+}
+
+
```
Comment (Shawntru) [inline src/scripts.js:160]: Whitespace fix

File: src/index.html (hunk 2 above)
Comment (saraho1123) [inline src/index.html:50]: I think you said you have an explanation to share with us about regex. I'm excited to see it!

File: src/scripts.js [hunk 8]
```diff
@@ -127,17 +211,26 @@ function makeRandomDate(userStorage, id, dataSet) {
 }
 
 function addHydrationInfo(id, hydrationInfo, dateString, userStorage, laterDateString) {
-  hydrationToday.insertAdjacentHTML('afterBegin', `<p>You drank</p><p><span class="number">${hydrationInfo.calculateDailyOunces(id, dateString)}</span></p><p>oz water today.</p>`);
-  hydrationAverage.insertAdjacentHTML('afterBegin', `<p>Your average water intake is</p><p><span class="number">${hydrationInfo.calculateAverageOunces(id)}</span></p> <p>oz per day.</p>`);
```
Comment (Shawntru) [inline src/scripts.js:131]: Where did these HTML inserts get moved to? Are we removing them completely?

File: src/scripts.js [hunk 9]
```diff
@@ -97,6 +118,69 @@ function getUserById(id, listRepo) {
   return listRepo.getDataFromID(id);
 }
 
+
+function windowOnClick(event) {
```
Comment (saraho1123) [inline src/scripts.js:122]: I like this windowOnClick function. I definitely plan to use this in the future!

File: src/fetch.js [hunk 10]
```diff
@@ -25,7 +25,61 @@ let requests = {
       .then(response => response.json())
       .then(data => data.sleepData)
       .catch(error => console.log(error))
-  }
+  },
```
Comment (Shawntru) [inline src/fetch.js:28]: Extra comma floating here

File: src/scripts.js [hunk 11]
```diff
@@ -97,6 +118,69 @@ function getUserById(id, listRepo) {
...
+    .then(value => {
+      hydrationData = value[0];
+      // let hydrationElementsToClear = q
+      hydrationToday.innerHTML = '';
+      startApp(userNowId)
+      // addDailyOuncesInfo(userNowId, currentHydrationRepo, todaysDate);
+    })
+}
+
+function submitNewActivity(userNowId) {
```
Comment (saraho1123) [inline src/scripts.js:173]: I like that you set these up to work on in the future ... which we have now done and will have in one of our next PRs!

File: src/Sleep.js [hunk 12]
```diff
@@ -1,4 +1,4 @@
-import sleepData from './data/sleep';
+// import sleepData from './data/sleep';
```
Comment (Shawntru) [inline src/Sleep.js:1]: Let's remove this whole line if we are commenting it out for this branch

**Review by Shawntru — CHANGES_REQUESTED** [review]

**Review by saraho1123 — APPROVED** [review]
Thanks for all the hard work Amanda!

File: src/Sleep.js (hunk 1 above)
Comment (ADavidson02) [reply src/Sleep.js]: Has been reverted to old code.

File: src/fetch.js (hunk 10 above)
Comment (ADavidson02) [reply src/fetch.js:28]: This has been deleted.

File: src/index.html (hunk 2 above)
Comment (ADavidson02) [reply src/index.html]: Removed second keyup

File: src/scripts.js (hunk 5 above)
Comment (ADavidson02) [reply src/scripts.js]: This has been removed.

File: src/scripts.js (hunk 6 above)
Comment (ADavidson02) [reply src/scripts.js]: This has been removed.

File: src/scripts.js (hunk 7 above)
Comment (ADavidson02) [reply src/scripts.js]: This has been removed.

File: src/fetch.js (hunk 4 above)
Comment (saraho1123) [reply src/fetch.js:30]: Shawn noticed that comma and I didn't. Thanks Shawn!

File: src/scripts.js (hunk 8 above)
Comment (ADavidson02) [reply src/scripts.js:131]: One has been moved to a new function addDailyOunces on line 216. The other is added back into the function 208.

File: src/Sleep.js [hunk 13]
```diff
@@ -15,11 +13,6 @@ class Sleep {
     return perDaySleepQuality.reduce((sumSoFar, data) => sumSoFar += data.sleepQuality, 0) / perDaySleepQuality.length;
   }
 
-  calculateDailySleep(id, date) {
-    const findSleepByDate = this.sleepData.find((data) => id === data.userID && date === data.date);
-    return findSleepByDate.hoursSlept;
-  }
-
```
Comment (ADavidson02) [inline src/Sleep.js:22]: @Shawntru This is the old code that I have changed back to.

File: src/Sleep.js [hunk 14]
```diff
@@ -1,5 +1,3 @@
-import sleepData from './data/sleep';
-
```
Comment (ADavidson02) [inline src/Sleep.js:2]: @Shawntru I have deleted this.

File: src/scripts.js [hunk 15]
```diff
@@ -44,33 +44,53 @@ const userMinutesThisWeek = document.getElementById('userMinutesThisWeek');
...
+
+
 Promise.all([receivedUserData, receivedActivityData, receivedHydrationData, receivedSleepData])
   .then(value => {
     userData = value[0]
     activityData = value[1];
     hydrationData = value[2];
     sleepData = value[3];
-    startApp();
+    startApp(0);
```
Comment (ADavidson02) [inline src/scripts.js:75]: This is now going to be outdated with what Sarah has done.

File: src/scripts.js [hunk 16]
```diff
@@ -44,33 +44,53 @@ const userMinutesThisWeek = document.getElementById('userMinutesThisWeek');
...
   const hydrationRepo = new Hydration(hydrationData);
   const sleepRepo = new Sleep(sleepData);
   const activityRepo = new Activity(activityData);
-  const userNowId = pickUser();
+  if (userIdCurrently !== 0) {
+    userNowId = userIdCurrently
+  } else if(userIdCurrently === 0){
+    userNowId = pickUser();
+  }
+  // userNowId = pickUser();
```
Comment (ADavidson02) [inline src/scripts.js:93]: This will now be outdated

**Review by ADavidson02 — COMMENTED** [review]
This still looks incorrect. I am working on fixing.

**Review by Shawntru — APPROVED** [review]

**Pull request merged**